src/App.tsx

```diff
@@ -1,10 +1,85 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { addTodo, getTodos, USER_ID } from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoItem } from './components/TodoItem';
+import classNames from 'classnames';
+import { ErrorMessage } from './types/ErrorMessage';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [activeTodos, setActiveTodos] = useState<Todo[]>([]);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [title, setTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>('');
+  const [loading, setLoading] = useState(true);
+  const [activeLink, setActiveLink] = useState('all');
+
+  const updateTodos = useCallback(
+    (link = activeLink) => {
+      switch (link) {
+        case 'all':
+          getTodos()
+            .then(allTodos => {
+              setTodos(allTodos);
+              setActiveTodos(allTodos.filter(todo => !todo.completed));
+              setCompletedTodos(allTodos.filter(todo => todo.completed));
+            })
+            .catch(() => setErrorMessage('Unable to load todos'))
+            .finally(() => setLoading(false));
+          break;
+        case 'active':
+          getTodos()
+            .then(allTodos => {
+              setTodos(allTodos.filter(todo => !todo.completed));
+              setActiveTodos(allTodos.filter(todo => !todo.completed));
+              setCompletedTodos(allTodos.filter(todo => todo.completed));
+            })
+            .catch(() => setErrorMessage('Unable to load todos'))
+            .finally(() => setLoading(false));
+          break;
+        case 'completed':
+          getTodos()
+            .then(allTodos => {
+              setTodos(allTodos.filter(todo => todo.completed));
+              setActiveTodos(allTodos.filter(todo => !todo.completed));
+              setCompletedTodos(allTodos.filter(todo => todo.completed));
+            })
+            .catch(() => setErrorMessage('Unable to load todos'))
+            .finally(() => setLoading(false));
+          break;
```

Comment:
It's enouth to fetch all todos once on page load, and filter them on switching activeLink.